src/rng.rs

```diff
@@ -1,28 +1,58 @@
 use crate::Vec;
-use algebra::{FromBytes, ToBytes, Field};
+use algebra::{FromBytes, ToBytes, to_bytes, Field};
 use std::{
     convert::TryInto,
-    marker::PhantomData
+    marker::PhantomData,
 };
 use digest::{generic_array::GenericArray, Digest};
 use rand_chacha::ChaChaRng;
 use rand_core::{RngCore, SeedableRng};
 use rand::Rng;
+use crate::error::Error;
+
+/// General trait for an updatable seed to be used to initialize a `FiatShamirRng`.
+pub trait FiatShamirRngSeed {
+    /// Output type of the seed, to be consistent with the seed type accepted
+    /// by the `from_seed` function of FiatShamirRng.
+    type FinalizedSeed: Sized + Clone;
+
+    /// Error type
+    type Error: std::error::Error + From<Error>;
+
+    /// Maximum number of elements allowed to be part of this seed
+    const MAX_NUM_ELEMENTS: usize;
+
+    /// Maximum length of each element that is part of the seed
+    const MAX_ELEMENT_LEN: usize;
+
+    /// Initialize this seed
+    fn new() -> Self;
+
+    /// Update this seed with a new element interpreted as a sequence of byte
+    fn add_bytes<'a, T: 'a + ToBytes>(&mut self, elem: &'a T) -> Result<&mut Self, Self::Error>;
+
+    /// Finalize this seed to the type needed by the corresponding FiatShamirRng impl
+    fn finalize(self) -> Self::FinalizedSeed;
+}
 
 /// General trait for `SeedableRng` that refreshes its seed by hashing together the previous seed
 /// and the new seed material.
-// TODO: later: re-evaluate decision about ChaChaRng
-pub trait FiatShamirRng: RngCore {
+pub trait FiatShamirRng: RngCore + Sized {
```

Comment:
Let us change/update the documentation of the trait.

Reply:
Done